build.gradle

```diff
@@ -49,6 +37,38 @@ dependencies {
 	}
 }
 
+sourceSets {
+	data {
+		compileClasspath += sourceSets.main.compileClasspath
+		compileClasspath += sourceSets.main.output
+		runtimeClasspath += sourceSets.main.compileClasspath
+		runtimeClasspath += sourceSets.main.output
+	}
+}
+
+def dataDir = project.file("$buildDir/datadump/")
+def dataCacheDir = new File(dataDir, "/.cache/")
+
+task generateData(type: JavaExec) {
```

Comment:
Will this not increase build time *significantly* if we have lots of data to generate?

Reply:
This should be cacheable. I will test more on the input and output of the task to make sure gradle can properly cache it.